### PR TITLE
Split EventSubscriber and ItemRepository

### PR DESCRIPTION
**Pull request by Bestils**

### CODE REVIEW

File: src/main/g8/src/it/scala/$package$/application/$if(add_caliban_endpoint.truthy)$SubsciberSpec.scala$endif$ [hunk 1]
```diff
@@ -0,0 +1,45 @@
+package com.example.application
+
+import com.example.ITSpec.ITSpec
+import com.example.domain._
+import com.example.infrastructure.flyway.FlywayProvider
+import zio.test.Assertion._
+import zio.test.TestAspect._
+import zio.test.{ suite, testM, _ }
+
+object deleteSpecForTests extends ITSpec(Some("items")) {
```
Comment (jczuchnowski) [inline src/main/g8/src/it/scala/$package$/application/$if(add_caliban_endpoint.truthy)$SubsciberSpec.scala$endif$:10]: This should be called `SubscriberSpec` like the file suggests.

File: src/main/g8/src/it/scala/$package$/application/$if(add_caliban_endpoint.truthy)$SubsciberSpec.scala$endif$ [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  def flippingFailure(value: Any): Exception =
+    new Exception(
+      "Flipping failed! The referred effect was successful with `" + value + "` result of `" + value.getClass + "` type!"
+    )
+
+  migrateDbSchema.useNow
+  val spec: ITSpec =
+    suite("Item Repository")(
+      //  def deletedEvents
```
Comment (jczuchnowski) [inline src/main/g8/src/it/scala/$package$/application/$if(add_caliban_endpoint.truthy)$SubsciberSpec.scala$endif$:25]: Can be removed.

File: src/main/g8/src/main/scala/$package$/infrastructure/$if(add_caliban_endpoint.truthy || add_server_sent_events_endpoint.truthy || add_websocket_endpoint.truthy)$EventSubscriber.scala$endif$ [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package $package$.infrastructure
+import $package$.domain.{Subscriber, ItemId }
+import zio._
+import zio.logging._
+import zio.stream.ZStream
+
+class EventSubscriber(env: Logging, deletedEventsSubscribers: Ref[List[Queue[ItemId]]]) extends Subscriber.Service {
```
Comment (jczuchnowski) [inline src/main/g8/src/main/scala/$package$/infrastructure/$if(add_caliban_endpoint.truthy || add_server_sent_events_endpoint.truthy || add_websocket_endpoint.truthy)$EventSubscriber.scala$endif$:7]: I believ `env: Logging` is not needed here.

**Review by jczuchnowski — CHANGES_REQUESTED** [review]

File: src/main/g8/src/it/scala/$package$/application/$if(add_caliban_endpoint.truthy)$SubsciberSpec.scala$endif$ (hunk 1 above)
Comment (Bestils) [reply src/main/g8/src/it/scala/$package$/application/$if(add_caliban_endpoint.truthy)$SubsciberSpec.scala$endif$:10]: Right,  that was an old name, I had to oversee that one.

File: src/main/g8/src/it/scala/$package$/application/$if(add_caliban_endpoint.truthy)$SubsciberSpec.scala$endif$ (hunk 2 above)
Comment (Bestils) [reply src/main/g8/src/it/scala/$package$/application/$if(add_caliban_endpoint.truthy)$SubsciberSpec.scala$endif$:25]: true

File: src/main/g8/src/main/scala/$package$/infrastructure/$if(add_caliban_endpoint.truthy || add_server_sent_events_endpoint.truthy || add_websocket_endpoint.truthy)$EventSubscriber.scala$endif$ (hunk 3 above)
Comment (Bestils) [reply src/main/g8/src/main/scala/$package$/infrastructure/$if(add_caliban_endpoint.truthy || add_server_sent_events_endpoint.truthy || add_websocket_endpoint.truthy)$EventSubscriber.scala$endif$:7]: Are you sure? In publishing, we have logging 
```Scala
def publishDeleteEvents(deletedItemId: ItemId): ZIO[Any, Nothing, List[Boolean]] =
    log.info(s"Publishing delete event for item \${deletedItemId.value}").provide(env) *>
      deletedEventsSubscribers.get.flatMap[Any, Nothing, List[Boolean]] { subs =>
        // send item to all subscribers
        log.info("subs" + subs)
        UIO.foreach(subs) { queue =>
          log.info("queue" + queue)```

**Review by jczuchnowski — APPROVED** [review]

**Pull request merged**